src/Libraries/CoreNodeModels/Selection.cs

```diff
@@ -56,6 +60,34 @@ private set
             }
         }
 
+        [JsonProperty("NodeType")]
+        public string NodeType
+        {
+            get
+            {
+                return "ExtensionNode";
+            }
+        }
+
+        [JsonProperty("InstanceId")]
+        public List<string> SelectionIdentifier
+        {
+            get
+            {
+                if (selectionIdentifier == null || selectionIdentifier.Count == 0)
+                {
+                    selectionIdentifier = selection.Select(GetIdentifierFromModelObject).Where(x => x != null).ToList();
+                }
+
```

Comment:
A small change is still needed here.  I believe this logic prevents the `selectionIdentifier` from updating once an initial value is stored.  It will appear as if the new selection has been updated in Dynamo background preview, however the initial selection id will serialize.

Reply:
you should write a test for that. Your own fault for bringing it up 😉

Reply:
haha probably for the best anyway :laughing: